### PR TITLE
drivers: clock_control: stm32: enable MCO configuration on L4 and F7

### PR DESCRIPTION
**Pull request by arbrauns**

### CODE REVIEW

**Comment by arbrauns** [comment]
It's worth noting that the MCO output will not work unless the GPIO pin is also configured appropriately through the devicetree, see #31912.

File: drivers/clock_control/Kconfig.stm32 [hunk 1]
```diff
@@ -62,25 +62,54 @@ config CLOCK_STM32_MCO1_SRC_EXT_HSE
 
 config CLOCK_STM32_MCO1_SRC_LSE
 	bool "LSE"
-	depends on SOC_SERIES_STM32F4X
+	depends on SOC_SERIES_STM32F4X || \
+		SOC_SERIES_STM32F7X || \
+		SOC_SERIES_STM32L4X
```
Comment (erwango) [inline drivers/clock_control/Kconfig.stm32:67]: Nit: Indentation.
```suggestion
		   SOC_SERIES_STM32F7X || \
		   SOC_SERIES_STM32L4X
```
Apply everywhere it's applicable

**Review by erwango — CHANGES_REQUESTED** [review]
This feature is half-baked and requires completion but current change doesn't do any harm, so I'm fine with it, aside from the small indentation nitpick

File: drivers/clock_control/Kconfig.stm32 (hunk 1 above)
Comment (arbrauns) [reply drivers/clock_control/Kconfig.stm32]: fixed.

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**